### PR TITLE
Adding the possibility to send an email with a template stored in mailgun

### PR DESCRIPTION
**Pull request by JannikZed**
Mailgun lets you predefine templates via the templates API. This pull request lets you send out these templates.

https://documentation.mailgun.com/en/latest/user_manual.html#templates

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -237,8 +237,14 @@ if (template && template instanceof MailgunTemplate) {
 			console.error(error);
 		});
 }
+```typescript
```
Comment (DrewImm) [inline README.md:240]: Please move the ```typescript block around sendOptions.template === 'TEMPLATENAME' on line 246

File: src/ts-mailgun.ts [hunk 2]
```diff
@@ -203,6 +203,8 @@ export class NodeMailgun {
 			};
 
 			message = Object.assign(message, sendOptions);
+			
+			if (!body || sendOptions?.template) delete message.html;
```
Comment (DrewImm) [inline src/ts-mailgun.ts:207]: Please put `delete message.html` on the next line and wrap in curly braces to match the rest of the code, like this:

```typescript
if (!body || sendOptions?.template) {
	delete message.html;
}
```

**Review by DrewImm — CHANGES_REQUESTED** [review]
Thank you for the pull request. Please adjust the formatting before I merge.

File: src/ts-mailgun.ts (hunk 2 above)
Comment (JannikZed) [reply src/ts-mailgun.ts]: fixed!

File: README.md (hunk 1 above)
Comment (JannikZed) [reply README.md]: fixed!

File: README.md [hunk 3]
```diff
@@ -237,13 +237,16 @@ if (template && template instanceof MailgunTemplate) {
...
-
 ```
 
+
 ### Sending with a Template stored in Mailgun
 To send via a pre-stored template, just leave the body empty and define the template name via:
-`sendOptions.template === 'TEMPLATENAME'`
+
+```typescript
+	sendOptions.template === 'TEMPLATENAME'
```
Comment (DrewImm) [inline README.md:247]: Should this be `===` or `=`?

**Comment by DrewImm** [comment]
Thank you for the work @JannikZed!

**Pull request merged**